// Copyright 2016 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package storage;

// This metadata is updated once per-binding of a LocalStorage StorageArea
// for use by LocalStorageControl.
message LocalStorageAreaAccessMetaData {
  // Serialized by base::Time::ToInternalValue().
  required int64 last_accessed = 1;
}

// This metadata is updated at the time a LocalStorage StorageArea is persisted
// to disk for a given StorageKey.
message LocalStorageAreaWriteMetaData {
  // Serialized by base::Time::ToInternalValue().
  required int64 last_modified = 1;
  required uint64 size_bytes = 2;
}
